opte-api/src/ip.rs

```diff
@@ -213,13 +271,69 @@ pub struct Ipv4Addr {
     inner: [u8; 4],
 }
 
+impl Ipv4Addr {
+    pub const ANY_ADDR: Self = Self { inner: [0; 4] };
+    pub const LOCAL_BCAST: Self = Self { inner: [255; 4] };
+
+    /// Return the bytes of the address.
+    pub fn bytes(&self) -> [u8; 4] {
+        self.inner
+    }
+
+    /// Return the address after applying the network mask.
+    pub fn mask(mut self, mask: u8) -> Result<Self, String> {
+        if mask > 32 {
+            return Err(format!("bad mask: {}", mask));
+        }
+
+        if mask == 0 {
+            return Ok(Ipv4Addr::ANY_ADDR);
+        }
+
+        let mut n = u32::from_be_bytes(self.inner);
+
+        let mut bits = i32::MIN;
+        bits = bits >> (mask - 1);
+        n = n & bits as u32;
+        self.inner = n.to_be_bytes();
+        Ok(self)
+    }
+
+    pub fn safe_mask(self, prefix_len: Ipv4PrefixLen) -> Self {
+        self.mask(prefix_len.0).unwrap()
+    }
+
+    /// Produce a `u32` which itself is stored in memory in network
+    /// order. This is needed for passing this type up to DTrace so
+    /// its inet_ntoa() subroutine works.
+    pub fn to_be(self) -> u32 {
+        // First we create a native-endian u32 from the network-order
+        // bytes, then we convert that to an in-memroy network-order
```

Comment:
Nit: spelling `in-memroy` -> `in-memory`

Reply:
Done.